nixos/modules/system/boot/networkd.nix

```diff
@@ -1979,7 +1962,30 @@ in
         };
       };
 
-      environment.etc."systemd/networkd.conf" = renderConfig cfg.config;
+    })
+  ];
+
+  stage2Config = mkMerge [
+    (commonConfig config)
+
+    (mkIf config.systemd.network.enable {
+
+      users.users.systemd-network.group = "systemd-network";
+
+      systemd.additionalUpstreamSystemUnits = [
+        "systemd-networkd-wait-online.service"
+        "systemd-networkd.service"
+        "systemd-networkd.socket"
+      ];
+
+      systemd.services.systemd-networkd = {
+        wantedBy = [ "multi-user.target" ];
+        restartTriggers = map (x: x.source) (attrValues (unitFiles config.systemd.network)) ++ [
+          config.environment.etc."systemd/networkd.conf".source
+        ];
+      };
+
+      systemd.services.systemd-networkd.aliases = [ "dbus-org.freedesktop.network1.service" ];
 
       networking.iproute2 = mkIf (cfg.config.addRouteTablesToIPRoute2 && cfg.config.routeTables != { }) {
```

Comment:
It seems like `cfg` doesn't exist here? I get an error when I try to use this:

```
error: undefined variable 'cfg'
         1994|
         1995|       networking.iproute2 = mkIf (cfg.config.addRouteTablesToIPRoute2 && cfg.config.routeTables != { }) {
             |                                   ^
         1996|         enable = mkDefault true;
```

Reply:
Ah, I hadn't been running the `nixos/tests/systemd-networkd.nix` test. Will do that going forward.